### ISSUE TITLE
Possible to call python script to notify of z-axis change, alert, or done? 

### ISSUE THREAD

**Issue by CptanPanic** [MaslowCNC/GroundControl#648]
I would like to send external notifications for state changes, and most flexible way would be to have ground control be able to setup external scripts for these states.  For example octoprint calls these events, and you can have it call any command line command when these events are triggered.  Could this be done or added to Ground Control?

Thanks

**Comment by CptanPanic** [comment]
I don't think it would need to wait for script or even expect an response from script.

**Comment by thormj** [comment]
Ooh... I like this one.  We could probably put hooks in for certain events (Z change), messages, or even hooks in the g-code eg`#{.pythonA take_calibration_picture.py ($x,$y,$z)}` (`#` makes it officially a comment, but `{.pythonA ` would tell Maslow to execute that script asynchronously.  Maybe `{.pythonS ` could be used for execute synchronously (wait for script to exit before sending next command -- if your command were to take time [like snap a picture and record the coordinates]).

One giant security hole, but... now Maslow & GC can work together at tasks without having to code them in each time.

**Comment by CptanPanic** [comment]
@thormj In octoprint, the M117 "Display Message" gcode is another alert. So this would be a more deterministic way.

**Comment by thormj** [comment]
I think I've got a handle on it if someone would like to assign it to me.  It'd be nice to get PR #636  in place first as this would go right in the same sections (the SerialThread).
